content/cucumber/reporting.md

```diff
@@ -235,6 +236,177 @@ Alternatively, you can specify a Java system property (in Maven, Gradle or other
+* `#skip_profile_information?`: A boolean indicating whether profile information should be skipped (`nil` by default)
+* `#snippet_generators`: An array of the snippet generators used, by default containing `Cucumber::Glue::Snippet::Generator`
+  objects
+* `#snippet_type`: A symbol indicating the type of snippet to be used (`:cucumber_expression` by default)
+* `#snippets?`: A boolean indicating whether snippets are to be generated for undefined steps (`true` by default)
+* `#source?`: A boolean indicating whether the file and line of the step definition should be printed with the output for
+  each step (`true` by default)
+* `#step_defs_to_load`: An array of the step definition files to be loaded for the test run
+* `#strict`: The `strict` setting to be applied to the test run (by default an instance of `Cucumber::Core::Test::Result::StrictConfiguration`)
+* `#support_to_load`: An array of the paths to support files to be loaded for the test run
+* `#tag_expressions`: An array of the tag expressions indicated at the command line or in the `cucumber.yml` config
+  file (empty by default)
+* `#tag_limits`: A hash of tag limits to be used (empty by default)
+* `#to_hash`: The options hash being used for the test run
+* `#wip?`: A boolean indicating whether the `--wip` option has been specified (`false` by default)
+* `#with_options`: A new config object merging existing options with a new options hash passed in as an argument
+
+### Event objects
+Every time an event is fired, an event object is passed to the corresponding handler. Each event object belongs to
+a class corresponding to its name; for instance, the `gherkin_source_read` event creates a
+`Cucumber::Events::GherkinSourceRead` object. Each of these objects provides a different API to access relevant
+data such as Gherkin source, feature and step names, and passed/failed status of a step or test case. All events
+inherit the methods `#attributes`, `#event_id`, and `#to_h` from the parent class.
+
+#### gherkin_source_read
+The `gherkin_source_read` event is fired after the contents of a new feature file have been read. It has two
+attributes:
+* `path`: The file path of the feature file that has been read
+* `body`: The raw Gherkin contained in that file, as a string
+
+#### step_definition_registered
+The `step_definition_registered` event is fired after each step definition has been registered. It has one
+attribute:
+* `step_definition`: The `Cucumber::Glue::StepDefinition` object that has just been registered. This object responds
+  to the following instance methods:
+    - `#backtrace_line`: The file, line, and step definition that will appear in a backtrace, as in:
+      `features/step_definitions/cuke_steps.rb:5:in "this step passes"`
+    - `#expression`: The expression used in the step definition, as in `"I have {int} cukes in my belly"` (when a
+      Cucumber expression is used) or `/I have (\d+) cukes in my belly/` (when a regular expression is used)
```

Comment:
Would this be anchored as regex? I'm not sure, but just thought I'd ask.

Reply:
@luke-hill I would think it would depend on whether it was anchored in the step definition the user wrote, which it generally should be. I'll go ahead and change this anyway.